R_Scripts/expanded_spectra-trait_kit_lma_plsr_example.R

```diff
@@ -22,12 +22,18 @@ closeAllConnections()   # close any open connections to files
 
 #--------------------------------------------------------------------------------------------------#
 ### Install and load required R packages
-list.of.packages <- c("devtools","readr","RCurl","httr","pls","dplyr","reshape2","here",
+list.of.packages <- c("devtools","remotes","readr","RCurl","httr","pls","dplyr","reshape2","here",
                       "ggplot2","gridExtra")  # packages needed for script
 # check for dependencies and install if needed
 new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]
-if(length(new.packages)) install.packages(new.packages)
-
+if(length(new.packages)) install.packages(new.packages, dependencies=c("Depends", "Imports",
+                                                                       "LinkingTo"))
+version_requirements <- c("3.3.2")
+if (!packageVersion("ggplot2") >= version_requirements[1]) {
+  remotes::install_version(package="ggplot2", version=paste0(">= ", version_requirements), 
+                           dependencies=c("Depends", "Imports", "LinkingTo"), upgrade="ask",
+                           quiet=TRUE)
```

Comment:
Just making sure I'm reading the documentation correctly, it asks the user before upgrading?

Reply:
yes it asks before upgrading dependencies because I didnt want to just assume someone would want to wait for that if not needed, which usually it isnt